packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-iam-fields.ts

```diff
@@ -0,0 +1,527 @@
+      expect(listPostsResult.data[listResultSetName].items).toEqual(expect.arrayContaining([expect.objectContaining(post)]));
+      // non-owner subscriber cannot update or delete any fields in post
+      const updatedPost = {
+        id: 'P-5',
+        title: 'My Post 5 updated',
+        owner: userName2,
+        subscriberList: [userName2, userName3],
+        likes: 10,
+        subscriberContent: 'Exclusive content 5 updated',
+      };
+      Object.entries(omit(updatedPost, 'id')).forEach(async (entry) => {
+        const updateInput = Object.fromEntries([['id', post.id], entry]);
+        await expect(async () => await postUser2Client.update(updateResultSetName, updateInput)).rejects.toThrowErrorMatchingInlineSnapshot(
+          expectedOperationError(updateResultSetName, 'Mutation'),
+        );
+      });
+      // cannot delete the post
+      await expect(
+        async () => await postUser2Client.delete(deleteResultSetName, { id: post.id }),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(deleteResultSetName, 'Mutation'));
+
+      // remove the post by private iam role
+      await postIAMPrivateClient.delete(deleteResultSetName, { id: post.id });
+    });
+
+    // helper functions
+    const constructModelHelper = (name: string, client): GQLQueryHelper => {
+      const createSelectionSet = /* GraphQL */ `
+        id
+        title
+        owner
+        subscriberList
+        subscriberContent
+        likes
+      `;
+      const updateSelectionSet = createSelectionSet;
+      const deleteSelectionSet = createSelectionSet;
+      const getSelectionSet = /* GraphQL */ `
+        query Get${name}($id: ID!) {
+          get${name}(id: $id) {
+            id
+            title
+            owner
+            subscriberList
+            subscriberContent
+            likes
+          }
+        }
+      `;
+      const listSelectionSet = /* GraphQL */ `
+        query List${name}s {
+          list${name}s {
+            items {
+              id
+              title
+              owner
+              subscriberList
+              subscriberContent
+              likes
+            }
+          }
+        }
+      `;
+      const helper = new GQLQueryHelper(client, name, {
+        mutation: {
+          create: createSelectionSet,
+          update: updateSelectionSet,
+          delete: deleteSelectionSet,
+        },
+        query: {
+          get: getSelectionSet,
+          list: listSelectionSet,
+        },
+      });
+
+      return helper;
+    };
+    const appendAmplifyInput = (schema: string, engine: ImportedRDSType): string => {
+      const amplifyInput = (engineName: ImportedRDSType): string => {
+        return `
+          input AMPLIFY {
+            engine: String = "${engineName}",
+          }
+        `;
+      };
+      return amplifyInput(engine) + '\n' + schema;
+    };
+    const omit = <T extends {}, K extends keyof T>(obj: T, ...keys: K[]) =>
```

Comment:
we can move the `appendAmplifyInput` and `omit` to common utilities for easy re-use.

Reply:
Will address this in the other PR